File: 1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.java
class Solution {
    public boolean uniqueOccurrences(int[] arr) {
        TreeSet<Integer> seen = new TreeSet<>();
        HashMap<Integer, Integer> hm = new HashMap<>();
        
        for (int i = 0; i < arr.length; i++) {
            hm.put(arr[i], hm.getOrDefault(arr[i], 0) + 1);
        }
        
        for (int freq : hm.values()) {
            if (seen.contains(freq)) return false;
            seen.add(freq);
        }
        
        return true;
    }
}

File: 1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.py
class Solution(object):
    def uniqueOccurrences(self, arr):
        """
        :type arr: List[int]
        :rtype: bool
        """
        seen=set()
        fre=Counter(arr)
        for i,f in fre.items():
            if(f in seen):
                return False
            seen.add(f)
        return True